tests/out/wgsl/operators.wgsl

```diff
@@ -7,7 +7,7 @@ fn builtins() -> vec4<f32> {
     let s2_: vec4<f32> = select(vec4<f32>(0.0, 0.0, 0.0, 0.0), vec4<f32>(1.0, 1.0, 1.0, 1.0), true);
     let s3_: vec4<f32> = select(vec4<f32>(1.0, 1.0, 1.0, 1.0), vec4<f32>(0.0, 0.0, 0.0, 0.0), vec4<bool>(false, false, false, false));
     let m1_: vec4<f32> = mix(vec4<f32>(0.0, 0.0, 0.0, 0.0), vec4<f32>(1.0, 1.0, 1.0, 1.0), vec4<f32>(0.5, 0.5, 0.5, 0.5));
-    let m2_: vec4<f32> = mix(vec4<f32>(0.0, 0.0, 0.0, 0.0), vec4<f32>(1.0, 1.0, 1.0, 1.0), 0.1);
+    let m2_: vec4<f32> = mix(vec4<f32>(0.0, 0.0, 0.0, 0.0), vec4<f32>(1.0, 1.0, 1.0, 1.0), 0.10000000149011612);
```

Comment:
I'm curious about this bit. This is `f32`, and the backend knows it's `f32`, so it sounds like there is a missing cast in the backend from f64 to f32 before printing it?

Reply:
Yes, there are two places where a cast would be necessary. I even tried out adding those casts, but then decided not to commit them for now since that makes it somewhat inconsistent with the other back-ends.
Would you like for me to add them to the wgsl backend? For the others I'd first have to check what float types are used, but they should all be `f32`s (or an equivalent type) as well, right?

Reply:
Though if that is the case, then the question is why an `f64` is used in the first place. Does one of the input languages allow for `f64`s? At least for that language (if output in it is supported) I wouldn't be able to add the cast.

Reply:
we just picked a biggest size. f64 are supported on some platforms.
Yes, you are right, we'd need this change on all textual backends...

Reply:
Should I add those casts to all textual back-ends then? I'd also add a TODO comment or something like that, since just casting blindly is a potential future footgun in that if an input lang starts supporting `f64`s, then the cast will still be there (unless it is manually removed)

Reply:
Wait, I'm confused. This change is about parsing WGSL literals. Why does it affect WGSL outputs?

Reply:
Because previously the `f32` literals were parsed as `f64`s and in my changes I switched to parsing as `f32`s and then casting to `f64` (so over-/underflow errors work).

Reply:
See https://github.com/gfx-rs/naga/blob/93db57c12b4a5eff48bdd00c494efa5ec89567ad/src/front/wgsl/mod.rs#L1120-L1123 for how it is done currently, without my changes. The parse call here uses the type from `ScalarValue::Float`, i.e. `f64`.

Reply:
thank you for explaining!